examples/haskell/example.hs

```diff
@@ -0,0 +1,16 @@
+
+import OpenCog.AtomSpace.Api
+import OpenCog.AtomSpace.Types
+import Control.Monad.IO.Class
+import Data.Default
+
+main :: IO ()
+main = runOnNewAtomSpace program
+
+program :: AtomSpace ()
```

Comment:
Hi. Yes, I plan to broaden the api to work with multiples atomspaces. In that case one would need to specify which atomspace wants to use for each action, so the atomspace instance wouldn't be hidden inside a monad.
Now:
   main = runOnNewAtomSpace $ do 
                                                            asAddNode  (Node ....)
                                                            asAddLink  (Link ....)

With multiples atomspaces:
  main = do
                   a1  <-  createAtomSpace
                   a2  <-  createAtomSpace
                   addNode a1 (Node ...)
                   addNode a2 (Node ...)
                   ....
I plan to offer both options. As it is more common to work with only one atomspace, the first approach is far more confortable because you avoid passing the atomspace instance as an argument each time for each action.
I will work on that and let you know!